File: src/main/java/world/guild/GuildSkill.java
package world.guild;

import util.packet.PacketWriter;

public class GuildSkill {

    private final short level;
    private final long expire;
    private final String name;

    public GuildSkill(short level, long expire, String name) {
        this.level = level;
        this.expire = expire;
        this.name = name;
    }

    public void encode(PacketWriter pw) {
        pw.writeShort(level);
        pw.writeLong(expire);
        pw.writeMapleString(name);
    }
}

File: src/main/java/world/guild/GuildSkill.kt
package world.guild

import util.packet.PacketWriter

class GuildSkill(private val level: Short, private val expire: Long, private val name: String) {

    fun encode(pw: PacketWriter) {
        pw.writeShort(level)
        pw.writeLong(expire)
        pw.writeMapleString(name)
    }
}